pkg/accounting/accounting.go

```diff
@@ -239,76 +260,43 @@ func (a *Accounting) Debit(peer swarm.Address, price uint64) error {
 }
 
 // Balance returns the current balance for the given peer
-func (a *Accounting) Balance(peer swarm.Address) (int64, error) {
-	peerBalance, err := a.getPeerBalance(peer)
+func (a *Accounting) Balance(peer swarm.Address) (balance int64, err error) {
+	err = a.store.Get(peerBalanceKey(peer), &balance)
+	if err == storage.ErrNotFound {
+		return 0, nil
+	}
```

Comment:
This can go inside `err != nil` (very minor optimization). Also, errors.Is would be preferred on the imported package error check.

Reply:
changed